### PR TITLE
Feature/restrict uploaded files

### PR DESCRIPTION
**Pull request by yoavrozner**

### CODE REVIEW

File: src/utils/drop.js [hunk 1]
```diff
@@ -1,110 +1,237 @@
...
+        return store.dispatch(
+            "onError",
+            new Error(i18n.t("errors.waitForUpload"))
+        );
     }
     const items = [...files];
-    store.commit(UploadSet.isUpload, true)
-    await Promise.all(items.map((file) => getEntries(file, parent, true).catch((err) => store.dispatch("onError", err))));
+    store.commit(UploadSet.isUpload, true);
+    const uploader = new UploadEntries(
```
Comment (nettashviro) [inline src/utils/drop.js:45]: Is there a reason you are constantly re-creating the same "upload" object and not defining it as a global object in the file?

File: src/utils/drop.js [hunk 2]
```diff
@@ -1,110 +1,237 @@
...
-                    (entry) =>
-                        !entry
-                            ? Promise.resolve()
-                            : getEntries(entry, res, isFirstFolder),
-                    { concurrency: NUM_OF_MAX_PROMISES }
-                );
-            }
-        };
-        await readEntries();
+        async function promisifiedReadEntries(entryReader) {
```
Comment (nettashviro) [inline src/utils/drop.js:192]: Change it to arrow function

File: src/utils/drop.js [hunk 3]
```diff
@@ -1,110 +1,237 @@
...
+            return await new Promise((resolve) =>
+                entryReader.readEntries((res) => resolve(res))
+            );
+        }
+
+        do {
+            entriesResult = await promisifiedReadEntries(entryReader);
+            entries.push(...Array.prototype.slice.call(entriesResult || [], 0));
+        } while (entriesResult.length);
```
Comment (nettashviro) [inline src/utils/drop.js:201]: nice!

**Review by nettashviro — APPROVED** [review]

**Comment by nettashviro** [comment]
please add docs

File: src/utils/drop.js (hunk 3 above)
Comment (yoavrozner) [reply src/utils/drop.js:201]: Thank you!

File: src/utils/drop.js [hunk 4]
```diff
@@ -66,6 +66,12 @@ class UploadEntries {
         this.maxNumOfFolders = maxNumOfFolders;
     }
 
+    /**
+     * A function that handles the uploads entries (of file or folder) functionality and handles errors.
+     * @param {DataTransferItem} file the file (or folder) item which will be uploaded.
```
Comment (nettashviro) [inline src/utils/drop.js:71]:
```suggestion
     * @param {DataTransferItem} file- file the file (or folder) item which will be uploaded.
```

File: src/utils/drop.js [hunk 5]
```diff
@@ -184,16 +217,20 @@ class UploadEntries {
         );
     }
 
+    /**
+     * Read all the files and folders from a given folder entry.
+     * @param {DirectoryEntry} entry a folder that is represented as a folder in the file system.
```
Comment (nettashviro) [inline src/utils/drop.js:222]:
```suggestion
     * @param {DirectoryEntry} entry - a folder that is represented as a folder in the file system.
```

**Review by nettashviro — APPROVED** [review]
please add '-' for the docs after the paramater name

**Pull request merged**